apollo-ios/Sources/Apollo/FieldSelectionCollector.swift

```diff
@@ -27,6 +29,10 @@ struct FieldSelectionGrouping: Sequence {
     fulfilledFragments.insert(ObjectIdentifier(type))
```

Comment:
Add a precondition check that a selection set is not added to both `fulfilledFragments` and `deferredFragments`.

Reply:
Done in https://github.com/apollographql/apollo-ios-dev/pull/217/commits/a92e0e6f57e8c02b74a105ba9d0fb13e2d16c70e.